### DISCUSSION TITLE
graphql/codegen/plugins/typescript/typescript-operations

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9511] [Docs Discussions]
# graphql/codegen/plugins/typescript/typescript-operations

Generate TypeScript operations from GraphQL queries. This plugin is based on the `typescript` plugin, but it generates operations instead of types.

https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-operations

**Comment by Azbesciak** [comment]
I do not  get why you just cannot allow to copy the input enum to the generated code, and I do not mean `enumValuesMap`, because for bigger project this is not a solution... That `enumValuesMap` even could be a solution, if it would take a function which receives original enum. But does it?

**Comment by ivoilic** [comment]
Using both optionals and Maybe<> seems like a weird default. Maybe it's just the API I'm consuming but it made anything option possibly null or undefined at the same time. Also Maybe is often applied in places that cause weird issues like for example the type of an array presumably because the array could be empty?! That in particular seems like a bug. I was forced to override maybe in my setup even though I would prefer to use it since the data does come back as null not undefined with the API I'm using.
